src/app/shared/model/story-distribution.model.ts

```diff
@@ -0,0 +1,74 @@
+import { Observable} from 'rxjs';
+import { HalDoc } from '../../core';
+import { BaseModel } from './base.model';
+import { REQUIRED } from './invalid';
+import { FeederEpisodeModel } from './feeder-episode.model';
+
+export class StoryDistributionModel extends BaseModel {
+
+  id: number;
+  kind: string = '';
+  url: string = '';
+
+  // external related models
+  episode: FeederEpisodeModel;
+
+  SETABLE = ['kind'];
+
+  VALIDATORS = {
+    kind: [REQUIRED()]
+  };
+
+  constructor(private series: HalDoc, story: HalDoc, storyDistribution?: HalDoc, loadRelated = false) {
+    super();
+    this.init(story, storyDistribution, loadRelated); // DO NOT load related by default
+  }
+
+  key() {
+    if (this.doc) {
+      return `prx.story.distribution.${this.doc.id}`;
+    } else if (this.series) {
+      return `prx.story.distribution.new.${this.series.id}`;
+    } else {
+      throw new Error('Cannot have a story-distribution outside of a series');
+    }
+  }
+
+  related() {
+    let episode = Observable.of(null);
+    if (this.isNew) {
+      episode = Observable.of(new FeederEpisodeModel(this.series, this.doc));
+    } else if (this.url) {
+      episode = this.doc.followLink({href: this.url}).map(edoc => {
+        return new FeederEpisodeModel(this.series, this.doc, edoc);
+      });
+    }
+    return {episode: episode};
+  }
+
+  decode() {
+    this.id = this.doc['id'];
+    this.kind = this.doc['kind'] || '';
+
+    // TODO: cms has non-auth'd urls
+    this.url = this.doc['url'] || '';
+    if (this.url && !this.url.match('/authorization/')) {
+      this.url = this.url.replace('/episodes/', '/authorization/episodes/');
+    }
+  }
+
+  encode(): {} {
+    let data = <any> {};
+    data.kind = this.kind;
+    return data;
+  }
+
+  saveNew(data: {}): Observable<HalDoc> {
+    throw new Error('Cannot directly create a story distribution');
+  }
+
+  swapNew(newModel: StoryDistributionModel) {
+    newModel.episode = this.episode;
```

Comment:
This is a bit funky.  When you save the StoryModel, CMS is going to create both the StoryDistribution and the FeederEpisode on the backend.  So both the StoryDistributionModel and the FeederEpisodeModel are reloaded/swapped in instead of POSTing anything new.